sdk/core/Azure.Core/tests/TestFramework/RecordedTestSanitizer.cs

```diff
@@ -40,11 +41,19 @@ public virtual byte[] SanitizeBody(string contentType, byte[] body)
         }
 
         public virtual string SanitizeVariable(string variableName, string environmentVariableValue) => environmentVariableValue;
+
         public virtual void SanitizeBody(RecordEntryMessage message)
         {
             if (message.Body != null)
             {
                 int contentLength = message.Body.Length;
+                if (message.Headers.TryGetValue(HttpHeader.Names.ContentLength, out string[] values) && values.Length > 0)
```

Comment:
Do we still need the logic in UpdateSanitizedContentLength after this change?

Reply:
Because what I think it was making sure is that we only update content length when the body length is changed, which won't be true anymore.

Reply:
We do need it, yes. I could've called the protected member myself, but the point of calling `base.Sanitize` was to not duplicate code as much as possible. But since, by then, Body.Length *had* changed (I changed it in my derivative) the code as written no longer changed Content-Length.

I'd prefer to just unconditionally overwrite it, in which case this check isn't needed. @tg-msft was there some reason you made updating Content-Length conditional? Seems it should just always be right which means we should unconditionally set it.

Reply:
I mean the length comparison logic (`originalLength != sanitizedLength`).

Reply:
I don't think so and opined internally we should remove that check, but wanted to find out why it was there in the first place. Seemed rather specific and didn't want to remove it without understanding why it was there. I certainly agree it's a lot cleaner.

Reply:
Oh, actually, it was @schaabs that wrote it. Scott, any reason you did a length check rather than just set it?

Reply:
I'm not sure but I seem to recall the sanitizer having this check before I refactored, and I merely moved this logic.

Reply:
Looks like I added this at https://github.com/Azure/azure-sdk-for-net/commit/d4599c1420ca398f0677594cea01e84c690b63d5#diff-41344154d605a673b24383a267c9b50c.  I think I was just being extra paranoid and it's safe to remove the check.